puppet/modules/slave/manifests/postgresql.pp

```diff
@@ -1,22 +1,38 @@
 # @api private
 class slave::postgresql {
-  Postgresql_psql {
-    cwd => '/',
-  }
+  # only CentOS slaves are used to run unit tests
+  if $::osfamily == 'RedHat' {
+    if $::operatingsystemmajrelease == '7' {
 
-  include ::postgresql::client
-  include ::postgresql::server
+			exec { 'remove postgresql':
+				command => "/bin/yum -y remove postgresql-server postgresql-client postgresql",
+				before => Class['postgresql::globals'],
+			}
 
-  # Simple known user/pass that will allow Jenkins to create its
-  # own databases when required
-  postgresql::server::role { 'foreman':
-    password_hash => postgresql_password('foreman', 'foreman'),
-    superuser     => true,
-    login         => true,
-  }
+      if $::operatingsystem == 'CentOS' {
+      	package { 'centos-release-scl-rh':
+          ensure => 'present',
+          before => Class['postgresql::globals'],
+        }
+      } else {
+        exec { 'enable SCL repo from RHUI':
```

Comment:
Can this be made idempotent? https://github.com/voxpupuli/puppet-rhsm uses `subscription-manager` instead of `yum-config-manager` but maybe that's a better way? They already did the hard work of making sure `subscription-manager` isn't called on every Puppet run.

Reply:
I can look at their for an example. These repositories are not controlled by subscription manager hence the need for yum.

Reply:
If they're already in a yum repo, you may get away with:
```puppet
yumrepo { 'rhel-server-rhui-rhscl-7-rpms':
  enabled => true,
}
```